src/zenml/zen_stores/migrations/versions/cbc6acd71f92_add_workspace_display_name.py

```diff
@@ -0,0 +1,58 @@
+"""add workspace display name [cbc6acd71f92].
+
+Revision ID: cbc6acd71f92
+Revises: 9e7bf0970266
+Create Date: 2025-03-12 19:38:57.126846
+
+"""
+
+import sqlalchemy as sa
+import sqlmodel
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "cbc6acd71f92"
+down_revision = "9e7bf0970266"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    """Upgrade database schema and/or data, creating a new revision."""
+    with op.batch_alter_table("workspace", schema=None) as batch_op:
+        batch_op.add_column(
+            sa.Column(
+                "display_name",
+                sqlmodel.sql.sqltypes.AutoString(),
+                nullable=True,
+            )
+        )
+
+    # Migrate existing workspace names to display_name
+    connection = op.get_bind()
+    meta = sa.MetaData()
+    meta.reflect(bind=connection, only=("workspace",))
+    workspace_table = sa.Table("workspace", meta)
+
+    for workspace_id, workspace_name in connection.execute(
+        sa.select(
+            workspace_table.c.id,
+            workspace_table.c.name,
+        )
+    ):
```

Comment:
Should we also ensure that existing workspace names are lowercase and have no special characters?

Reply:
This has a near-zero chance of happening. I already blocked the DB upgrade in the previous PR if there is more than one workspace present. And it's very improbable that someone would actually use a weird name for their default workspace, given that it wasn't possible to manage them with the CLI and dashboard before.